src/client.js

```diff
@@ -369,6 +371,30 @@ export class Saturn {
     }
   }
 
+  _sortNodes (nodes) {
+    // Determine the maximum distance for normalization
+    const maxDistance = Math.max(...nodes.map(node => node.distance))
+
+    // These weights determine how important each factor is in determining
+    const distanceImportanceFactor = 0.8
+    const weightImportanceFactor = 1 - distanceImportanceFactor
+
+    nodes.sort((a, b) => {
```

Comment:
nit: `nodes.sort` is mutative. If you want the function to be pure, you should slice the list before sorting, like `return nodes.slice().sort()`

Reply:
Thanks @guanzo, great to learn :)

Reply:
well now you broke it because you're sorting the slice (new array) but returning the old array